tensorflow.go

```diff
@@ -0,0 +1,172 @@
+package tensorflow
+
+import (
+	"encoding/binary"
+	"fmt"
+	"reflect"
+	"unsafe"
+
+	"github.com/golang/protobuf/proto"
+	tf "github.com/tensorflow/tensorflow/tensorflow/go"
+)
+
+type Session struct {
+	session tf.TF_Session
+}
+
+func NewSession() (*Session, error) {
+	status := tf.TF_NewStatus()
+	return &Session{
+		session: tf.TF_NewSession(
+			tf.TF_NewSessionOptions(),
+			status,
+		),
+	}, statusToError(status)
+}
+
+type Tensor struct {
+	tensor tf.TF_Tensor
+	buf    []byte
+}
+
+type TensorShape [][]int64
+
+var (
+	TensorShapeScalar = TensorShape{{1}}
```

Comment:
could you comment why you're using this as a var?  I assume it's just optimization for Constant?